bundles/org.openhab.binding.panamaxfurman/src/main/java/org/openhab/binding/panamaxfurman/internal/PanamaxFurmanAbstractHandler.java

```diff
@@ -0,0 +1,219 @@
+import org.openhab.binding.panamaxfurman.internal.protocol.event.PanamaxFurmanInformationReceivedEvent;
+import org.openhab.binding.panamaxfurman.internal.transport.PanmaxFurmanConnector;
+import org.openhab.binding.panamaxfurman.internal.util.TimeInterval;
+import org.openhab.core.config.core.Configuration;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link PanamaxFurmanHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Dave Badia - Initial contribution
+ */
+@NonNullByDefault
+public abstract class PanamaxFurmanAbstractHandler extends BaseThingHandler
+        implements PanamaxFurmanCommunicationEventListener {
+    private final Logger logger = LoggerFactory.getLogger(PanamaxFurmanAbstractHandler.class);
+
+    private static final long AVOID_DUPLICATE_REQUESTS_WITHIN_MILLIS = TimeUnit.SECONDS.toMillis(3);
+    private static final Map<String, Long> REQUEST_LAST_SENT_AT = new ConcurrentHashMap<>();
+    private final ProtocolMapper protocolMapper;
+    private @NonNullByDefault({}) PanmaxFurmanConnector connector;
+    @SuppressWarnings("rawtypes")
+    private @Nullable ScheduledFuture backgroundJob;
+
+    public PanamaxFurmanAbstractHandler(Thing thing, ProtocolMapper protocolMapper) {
+        super(thing);
+        this.protocolMapper = protocolMapper;
+    }
+
+    /**
+     * Request the implementation to build a connector for the protocol
+     */
+    protected abstract PanmaxFurmanConnector buildConnector(Configuration genericConfig);
+
+    /**
+     * The name of this device connection
+     */
+    protected abstract String getConnectionName();
+
+    /**
+     * The interval at which the device should be polled for status updates. Will vary depending on whether the
+     * underlying protocol supports push updates or not
+     */
+    protected abstract TimeInterval pollInterval();
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+        this.connector = buildConnector(getConfig());
+        logger.debug("starting scheduled status now then repeating {}", getConnectionName());
+        // Request ID info immediately
+        connector.sendRequestToDevice(protocolMapper.buildQueryString(PowerConditionerChannel.ID, null));
+        // Query the state every 5 minutes
+        // A request for outlet 1 will return status for all outlets
+        backgroundJob = scheduler.scheduleAtFixedRate(() -> requestStatusOf("outlet1#power"), 0,
+                pollInterval().duration(), pollInterval().unit());
```

Comment:
pollInternal is abstract because different protocols will have very different requirements.  Currently only telnet is implemented, but when we add http the polling will have to happen more frequently to keep the state updated.  This logic helps to clarify that the polling internal will change depending on implementation.